### ISSUE TITLE
JPEG XL image format

### ISSUE THREAD

**Issue by Schweinepriester** [web-platform-tests/interop#700]
### Description

Same as last year, see https://github.com/web-platform-tests/interop/issues/430 and the year before https://github.com/web-platform-tests/interop/issues/176.

### Specification

https://jpeg.org/jpegxl/documentation.html; ISO/IEC 18181

### Additional Signals

- https://caniuse.com/jpegxl
- Supported by Safari
- Firefox
   - https://github.com/mozilla/standards-positions/issues/522
   - with recent addition https://github.com/mozilla/standards-positions/pull/1064
- Chrome: https://chromestatus.com/feature/5188299478007808
- WPT Tests: https://wpt.fyi/results/jpegxl

**Comment by Mouvedia** [comment]
see also https://issues.chromium.org/issues/40168998

**Comment by conradsrc** [comment]
[Here's a different, related Chromium ticket](https://issues.chromium.org/issues/40270698) with a lot of activity about reopening the JXL issue.

I'll also copy a comment I recently left there that summarizes some recent coordination between the JPEG-XL team at Google Research, the Firefox team at Mozilla, and the [jxl-oxide](https://github.com/tirr-c/jxl-oxide) maintainer.

>Adding onto the [recent news](https://github.com/mozilla/standards-positions/pull/1064) that Mozilla is considering a Rust implementation of JPEG-XL, [here's a comment from someone on the Google Research JPEG-XL team](https://github.com/web-platform-tests/interop/issues/430#issuecomment-1776762287) from last October's Interop discussion where they confirm that they could make a high-quality Rust decoder within several months, as well as the fact that they've already made the Chromium interfacing code for the C++ reference implementation.
>
>>If having a high quality Rust decoder implementation would arise as the only gating factor for choosing JPEG XL into interop 2024, the JPEG XL developer team in Google Research can deliver such by the end of Q2 2024. We have been successfully developing and maintaining Brotli and WOFF2 codecs for almost 10 years and would be able to do this too, safely and professionally, as well as supporting for the HDR needs there (such as ultra HDR, PQ, HLG and others).
>>
>>We can also create the interfacing code for the browsers, including Chrome, Edge and Mozilla. We already have a version of C++ JPEG XL reference implementation interface code for Chrome and Edge ready for deployment, and could do the same for the possible Rust version.
>
>[Here's a follow-up comment 3 weeks later](https://github.com/web-platform-tests/interop/issues/430#issuecomment-1811972676) from Jon Sneyers, the JPEG-XL lead. Jon confirms the already-existing jxl-oxide Rust decoder fully confirms to the spec and passes all conformance tests:
>
>>We have tested conformance of the jxl-oxide decoder (which is implemented in Rust) and **it is a fully conforming alternative implementation of JPEG XL**. It correctly decodes all conformance test bitstreams and passes the conformance thresholds described in ISO/IEC 18181-3.
>
>Three days ago, the owner of jxl-oxide [confirmed they were coordinating with the core JPEG-XL devs](https://github.com/tirr-c/jxl-oxide/issues/335#issuecomment-2328649512) to figure out how to potentially incorporate his project/efforts:
>
>>I'm in contact with some of the core devs in the JPEG XL community Discord server, and whether to use jxl-oxide as a base for the support in Firefox is yet to be decided (more work is needed, especially to match performance requirements, at least).

Meanwhile, Apple has continued to develop its JXL support on their product lines. Earlier today, [Apple confirmed](https://petapixel.com/2024/09/18/why-apple-uses-jpeg-xl-in-the-iphone-16-and-what-it-means-for-your-photos/) iPhone 16 Pros will be able to use JPEG-XL compression for ProRAW photos:

>**Apple and its various software iterations have supported JPEG XL for at least a year**, including in Finder, Preview, Final Cut Pro, Pages, Photos, Mail, Safari, and more. Adobe has also supported the format for a while, including in Adobe Camera Raw and Lightroom Classic.
>
>As for why it is including JPEG XL in the iPhone 16 Pro, Apple tells PetaPixel that the format promises two primary benefits over standard JPEG format: improved image quality and better compression performance. **If there’s a 32MB JPEG image, that same photo will be 24MB in lossless JPEG XL and, even more impressively, about five megabytes in perceptually lossless format.**
>
>Apple has wrapped JPEG XL photos inside a DNG container, enabling ProRAW files to retain their flexibility while being significantly smaller — up to nearly five times smaller. 

![image](https://github.com/user-attachments/assets/be0671bb-4e12-4d5e-b872-704b9ceb663c)

**Comment by hachahbar** [comment]
This year we must demand an explanation if the proposition is turned down again. No more backroom shenanigans.

**Comment by gordonfreeman01** [comment]
> This year we must demand an explanation if the proposition is turned down again. No more backroom shenanigans.

Agreed, unless there's a valid explanation (unlikely) this format not being accepted through this platform is wholly unacceptable when it clears the due process.

**Comment by andrews05** [comment]
As much as I would love to see Chrome forced to implement JPEG XL, I'm afraid these comments making "demands" are based on a misunderstanding of who and what Interop is. The Interop team is based on voluntary participation from the browser makers and has no authority to force anyone to do anything. If the Interop team approves a proposal that a browser maker has no intention of implementing, then it still won't be implemented and the process serves no purpose.

See the [Team Charter](https://github.com/web-platform-tests/interop/blob/main/charter.md) for more info. A key quote:

> The team makes decisions based on consensus. A decision has consensus if it has support from at least two participating organizations and no opposition.

Basically, if Chrome opposes JPEG XL then it won't be approved, period.

**Comment by hachahbar** [comment]
How do we know that Chrome blocks the adoption of JPEG XL within Interop? We don't know _for sure_ who blocks it because the whole thing is opaque for us plebeians, so you can only assume. This is what I am talking about. We need transparency, if the chrome team blocks JXL, it needs to be known. The more a company abuses its position the more likely for it to be considered a monopoly and broken up. Last month, Google's search engine was found to be an illegal monopoly and the judge is considering forcing Google to spin off the Chrome browser. Another anti-trust lawsuit will look into Google's ad business this month. Hopefully this will knock some sense into them.

**Comment by Patronics** [comment]
> ### Description
> Same as last year, see #430 and the year before #176.

Given some comments last year suggested including the description in that comment thread, rather than just linking to a previous discussion, perhaps someone should do the same this year, ensuring there's a clear explanation of what JPEG-XL is and why it matters?

**Comment by KParthSingh** [comment]
For those claiming there isn’t enough interest or demand from the ecosystem for JPEG XL, here's a reality check.
![F2icFW6XsAAGiFQ](https://github.com/user-attachments/assets/2d56c2e7-e410-46d9-8ad6-e5b19e0d2352)

**Comment by BearCooder** [comment]
During last years biggest performance meetup, PerfNow, at the pre-conference meetup, there was a very convincing song live played. **Give JXL a Chance** - https://www.youtube.com/watch?v=VJaa1Le4W7c

Also unrelated to the song I think this sums it up pretty good:

![jpge xl](https://github.com/user-attachments/assets/e55b2778-3f82-4364-a926-6a6b27fa162e)

**Comment by jyrkialakuijala** [comment]
Dear JPEG XL community, let's focus on the future of JPEG XL in Interop 25

I have two suggestions to help move forward:

Focus on Solutions, Not Blame: The Interop team works hard within their process. Let's not dwell on last year's decision.
Provide Context for Newcomers: Copy relevant information from last year's discussions and other sources for those unfamiliar with the issue. Feel free to reuse my old positions from https://github.com/mozilla/standards-positions/issues/522#issuecomment-853919129
Describe JPEG XL benefits broadly, especially those functions that are not covered by existing modern solutions.

Since the reason for JPEG XL's exclusion in Interop 24 is unclear,  we need a general overview of the format rather than address specific issues.

There might be a connection to the Rust decoder issue we discussed last year. To address this, the JPEG XL developers at Google are working on a new Rust implementation. You can find updates on this project at [link to GitHub repository](http://github.com/libjxl/jxl-rs).

Key Points:

- We want to be constructive in moving forward with JPEG XL for Interop 25.
- Broad and detailed background information will be helpful for those new to the discussion.
- A new Rust decoder is under development by the JPEG XL team at Google, which could be relevant to Interop 25.

**Comment by mtom55** [comment]
To the JPEG XL community:

One of the key issues that affect JPEG XLs inclusion is the lack of transparency in the Interop process.  We (Open Web Advocacy) believe that we should require this transparency. 

We are pushing for transparency via this issue [Interop Lack of Transparency & Accountability](https://github.com/web-platform-tests/interop/issues/888)

**Comment by o-t-w** [comment]
Firefox’s reservations sound entirely reasonable. I don’t see any point in making JPEG XL part of Interop until the new Rust decoder is proven to be a success. https://github.com/mozilla/standards-positions/pull/1064

**Comment by jacob-willden** [comment]
As far as Rust decoders for JPEG-XL go, there's not only [jxl-rs](https://github.com/libjxl/jxl-rs), but also [jxl-oxide](https://github.com/tirr-c/jxl-oxide). I'm not sure which one is more complete at this point, but it appears that either could be used in the future by Mozilla to support JPEG-XL.

**Comment by conradsrc** [comment]
[DICOM](https://en.wikipedia.org/wiki/DICOM), or Digital Imaging and Communications in Medicine is a worldwide standard for the storage and transfer of medical imaging data, with "a near universal level of acceptance among medical imaging equipment vendors and healthcare IT organizations". They just [updated their standard](https://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_A.4.12) to adopt JPEG-XL as a payload codec.

[Here are some slides](https://www.dicomstandard.org/news-dir/current/docs/sups/sup232-slides.pdf) covering the decision. Some snippets:

* Ability to forward migrate into JPEG XL without any loss of fidelity with
existing JPEG, as a safe migration option with full backwards compatibility
* As DICOMweb response is multiframe at much higher quality than gif
* As lossless web response, is much smaller than PNG and easier to decode
* Works well for both natural and synthetic images
* No visual difference from what is seen, with 50-60% smaller
* Profiling results show that JPEG XL is fast enough for decoding/encoding, but
is not as fast as JPEG
* WSI scanner vendors are interested in trying to generate JPEG XL directly
* Reduction in artifacts is important for algorithms processing, and is better with
JPEG XL
* Streaming of data to both algorithms and viewers is important and better with
JPEG XL
* Can render portions of a tile that are important first (progressive)
* May well be able to reduce quality to reduce size further without reducing
machine or user quality as JPEG XL has lots of options with very few artifacts
* JPEG XL is a logical extension to baseline JPEG, especially if easily
supported in browsers.

They of course mention that incomplete web browser support is a downside but have gone ahead anyway. And based on what we've heard from Mozilla recently (detailed above), there's a fair chance that the "incomplete" support will be "most Chromium-based browsers" in the near future.

Also of note from the standard:

> A JPEG Baseline image losslessly re-coded to JPEG XL is not a derived image unless the original JPEG image was a derived image.

[Here's an IEEE proposal](https://ieeexplore.ieee.org/document/10205928) from a conference in June 2023 that was advocating for DICOM to adopt JXL's lossless compression as "a universal replacement for medical file size reduction."

**Comment by tif-calin** [comment]
Mozilla's concern is stated clearly

> the increased attack surface of the reference decoder (currently behind a pref in Firefox Nightly), which weighs in at more than 100,000 lines of multithreaded C++

Chrome's position is also often misrepresented/reduced to just "not enough interest" (blatantly false ofc), but also

> The new image format does not bring sufficient incremental benefits over existing formats to warrant enabling it by default

I think Mozilla's position is entirely reasonable. They are working with Google to develop a rust-based decoder that could resolve their concerns. I don't think it's fair to push them when they're actively looking into a solution that addresses their concerns

Chrome's position needs further elaboration and could probably be argued against.

**Comment by Best-HeyGman** [comment]
This issue is even more important than people commonly think. There will be a successor to JPEG, but JPEG-XL is the only candidate that allows a lossless transcoding from JPEG to JPEG-XL with a ~20% size reduction (according to my own tests).

Images on the web have already degraded due to generation loss from recompression. With WEBP, AVIF or HEIC you don't have any other choice than to recompress again with a loss, if you want to get any size reduction when switching to these formats.

**I do think we owe it to the people that come after us to choose the successor to JPEG wisely and to consider the impact that our choice has on the preservation of images that already exist right now. It would be a shame to see them reduced to a porridge of pixels by the time they would be used to teach history lessons.**

**Comment by AngelaDMerkel** [comment]
> Images on the web have already degraded due to generation loss from recompression. With WEBP, AVIF or HEIC you don't have any other choice than to recompress again with a loss, if you want to get any size reduction when switching to these formats.

One of the advantages of JXL is that even in JXL to JXL recompression, the standard is designed to limit generation loss as we see in the JPEG to JPEG recompression. This is a huge advantage in some kinds of web communities

**Comment by jonnyawsom3** [comment]
Depending on the site, data usage can be limited by simply partially downloading JXL files.
For thumbnails, 0.1% is enough to load a file depending on the image. 20% for a low quality preview, ect. All without any extra data being transferred, only the rest of the file being requested.
Other formats require multiple different sizes or qualities to be stored in excess, adding processing time that you get for free with JPEG XL.

Legacy clients can still be served transcoded files, as the overhead is under 60ms to revert back to JPEG, further reducing storage costs.

Google's own saliency demo shows that regions of images can be prioritised, improving responsiveness with the aforementioned partial requests. Showing the focus first and the background last.

It's also redundant to data loss or corruption, albeit not currently enabled in libjxl. The image can still be rendered with missing AC or DC blocks, so long as the header is intact.
(If AC is corrupt, the underlying DC colors will be shown. If DC is corrupt, the overlayed AC texture will still be applied to a black/estimated DC color)

I think others have already said the common use cases and examples. The Spec is also being updated to allow UltraHDR transcoding, further futureproofing the format and retaining backwards compatibility.

**Comment by mo271** [comment]
Good news! The [jxl-rs](https://github.com/libjxl/jxl-rs) project (a safe and fast JPEG XL decoder implementation in Rust) is progressing well. We are currently on track to deliver the following milestones:

End of February 2025: Initial decoding capabilities and a preliminary API.
~April 2025: Aiming for a conforming decoder implementation, fully compliant with the JPEG XL specification.
~July 2025: Critical code paths fully SIMDified and with a finalized API.
This anticipated timeline should allow jxl-rs to be ready for browser integration in alignment with Interop 2025 goals.

We appreciate the community's continued interest and support.  Please remember to treat each other with respect in all project discussions. A positive and collaborative environment is crucial for the success of JPEG XL.

**Comment by mchv** [comment]
On the last day of this year, I thought it would be helpful to share an update from [theguardian.com](https://www.theguardian.com/) as we have publicly voiced support for the format, and reiterate that we would like browser engineering teams to focus on supporting this image format.

Earlier this year, in the summer, we started serving JPEG XL images to our readers when supported by their browsers. We were the first customer of our CDN image service to do so, and you can see impact on their global requests breakdown by format for them:

![Screenshot 2024-12-31 at 13 40 03](https://github.com/user-attachments/assets/04c7994c-6c4f-4e13-9e24-71e336e369db)

We are impatiently waiting for an increased support by browsers and image processing software, as we aim to be able to use JPEG XL end-to-end from photographer’s camera to end users. We therefore consider [Mozilla's new position ](https://github.com/mozilla/standards-positions/pull/1064)an excellent news! 

We would also appreciate it if Webkit could support [progressive decoding](https://bugs.webkit.org/show_bug.cgi?id=272350) and in general better support [progressive images rendering while loading](https://bugs.webkit.org/show_bug.cgi?id=249113).

Finally as we’ve seen with AVIF, codecs improve over time. We think the best way for them all to reach their full potential is to allow them to compete by ensuring fair support.

**Comment by o-t-w** [comment]
@mo271 will it include support for animated JPEG XL? It’s a shame that the Safari implementation doesn’t include animation.

**Comment by kanashimia** [comment]
@o-t-w jxl-rs can already read animations to some basic degree.
But Firefox and Safari use libjxl AFAIK, which can obviously work for decoding animations, they just don't integrate it well enough, which is a different problem. Firefox and Safari also had problems with transparency and progressive decoding last time I checked. So it will be up to the consumers of the library to integrate it with all of its features.
This is based on my understanding, can correct me on this.

**Comment by nekohayo** [comment]
@Schweinepriester it might be worth editing the description above to note that, in addition to Safari (WebKit), JPEG XL is also working out-of-the-box in WebKitGTK-based browsers and applications, such as [Epiphany](https://en.wikipedia.org/wiki/GNOME_Web) and others (ex: desktop email clients / feed readers / social media clients that use WebKitGTK, and presumably browsers in the embedded space, etc.). That information is unfortuntely [not reflected](https://github.com/Fyrd/caniuse/issues/6807) on the CanIUse website.

**Comment by mo271** [comment]
> @mo271 will it include support for animated JPEG XL? It’s a shame that the Safari implementation doesn’t include animation.
> 
> There are [use cases](https://jakearchibald.com/2024/video-with-transparency/) that require both animation and transparency. Hopefully Interop will be for all features of JPEG-XL. It's partly its feature set that makes JPEG-XL a worthwhile format in the first place.

Yes, we plan to include support for animated JPEG XL images.

**Comment by wpt-interop** [comment]
Thank you for proposing JPEG XL image format for inclusion in Interop 2025.

We wanted to let you know that this proposal was not selected to be part of Interop this year.

On behalf of the entire Interop team, thank you for submitting this proposal for consideration. We got many more proposals than we could include in this year's project, necessitating some difficult choices. Please note this should not be taken as a comment on the technology as a whole, or our willingness to consider it for Interop in the future. We appreciate the work you put into your proposal, and would welcome your participation in future rounds of Interop.

For an overview of our process, see [proposal selection](https://github.com/web-platform-tests/interop/blob/main/2025/selection-process.md). Thank you again for contributing to Interop 2025.

_Posted on behalf of the Interop team._

**Comment by eylenburg** [comment]
Would it be possible to see who (of the organizations who have decision power) voted for or against this proposal?

**Comment by Best-HeyGman** [comment]
Well, 33 year old jpeg it is then for another year. 🤦

**Comment by goodusername123** [comment]
(**Disclaimer:** I am not affiliated in anyway with the Interop project besides leaving a comment on last year's public proposal. additionally I apologize for the poor grammar and such likely present within this whole post/comment.)

From what I understand the point of the Interop project isn't to make sure things that _should_ be supported by web browsers are supported it's to make sure that things _already_ supported or agreed upon by all browsers work properly and consistently across said browsers, For example look at all the other useful and highly rated proposals from last year (and likely this year as well though I haven't checked this year's ones) that where rejected and this becomes more apparent/clear.

However I do think that at the very least this proposal and the one from last year are useful (even if destined to be rejected) in the sense that they show how sought after JPEG XL is as a feature by those both inside and outside the Web dev sphere/space to eyes/people (some of which are close to the spaces where web standards get made or pushed forward!) who otherwise would've not known about anything revolving around JPEG XL or it's current situation.

Over the years I have sadly seen some very bad faith decisions by people/teams in high positions of power that has lead to the place where some things are today however AFAIK the majority if not all of the people working at/on the Interop project are not to blame and I would say that anger and disdain people have relating to this situation should not be directed at Interop project staff/volunteers. I remember last year seeing people end up blindly directing anger towards people who where/are in support JPEG XL just because they where apart of the Interop team/project in some form.

I understand the anger around JPEG XL not being present across web browsers as I have personally been following the format and the situation around it for years and have watched it be unfairly excluded from various places due to reasons that have no relation to effort around technical implementation and even in the face of multiple large entities/companies both inside and outside the Web sphere voicing their support for it (some even explicitly stating their want of support for it over  other modern competing formats). And these things are very saddening to see as someone who loves studying and researching how people interact with the world and tech and how to make the Web and software in general a better place for both those who understand it and those who don't (and also for the environment as well).

**Comment by LifeIsStrange** [comment]
From what I remember, the removal of JPEG XL from chromium was done by a rogue manager without coherent justification, out of politics and public major conflict of interests/mafia. This guy was a manager (founder?) at the webp codec though the conflict of interest even if driven by jealousy, is not in itself even coherent, since it was made public that webp 2.0 development has been abandonned. The fact that a single rogue agent can do the equivalent of a state coup on the browser interop and standard adoption, is an extremely worrying precedent making the web a closed, undemocratic and unrepresentative oligarchy. The silence from other decision makers is deafening.

**Comment by ach4ndi** [comment]
> From what I remember, the removal of JPEG XL from chromium was done by a rogue manager without coherent justification/ straw man, out of politics and public major conflict of interests/mafia. This guy was a manager (founder?) at the webp codec though the conflict of interest even if driven by jealousy, is not in itself even coherent, since it was made public that webp 2.0 development has been abandonned. The fact that a single rogue agent can do the equivalent of a state coup on the browser interop and standard adoption, is an extremely worrying precedent making the web de facto a closed, undemocratic and unrepresentative oligarchy. The silence from other decision makers is deafening and public treason of the interests of mankind.

Maybe he just didnt like why jpeg xl so fast get acceptable from user and developers, when comparing to webp, need 1 decades, still no one ever used it except is being forced as modern web image fornat.

**Comment by mgenware** [comment]
Let’s save the world tons of resources and energy by reducing all JPEGs by 20% on web losslessly YES LOSSLESSLY!!! with JX.
AL
Apple: I'm one of the biggest mobile device manufacturers. I feel obliged to support that.
Google: Nope.
(EU should enact another law to enforce better formats and f*** those stupid companies Infested by politics)

**Comment by prlbr** [comment]
> Thank you for proposing JPEG XL image format for inclusion in Interop 2025.
> 
> We wanted to let you know that this proposal was not selected to be part of Interop this year.
> 
> On behalf of the entire Interop team, thank you for submitting this proposal for consideration. We got many more proposals than we could include in this year's project, necessitating some difficult choices. Please note this should not be taken as a comment on the technology as a whole, or our willingness to consider it for Interop in the future. We appreciate the work you put into your proposal, and would welcome your participation in future rounds of Interop.
> 
> For an overview of our process, see [proposal selection](https://github.com/web-platform-tests/interop/blob/main/2025/selection-process.md). Thank you again for contributing to Interop 2025.
> 
> _Posted on behalf of the Interop team._

It's unfortunate that Interop does not publish specific reasons for not selecting a proposal. More transparency on why a proposal was not selected in a given year could help finding a constructive path forward and would reduce the risk of misleading speculations.
